GitUI/CommandsDialogs/CommitDialog/CommitTemplateItem.cs

```diff
@@ -110,7 +113,28 @@ private static CommitTemplateItem[] DeserializeCommitTemplates(string serialized
 
             return commitTemplateItem;
         }
-    
+
+        private static readonly Dictionary<string, Func<string>> RegistratedTemplates = new Dictionary<string, Func<string>>();
+        [NotNull]
+        public static CommitTemplateItem[] LoadFromRegistrated()
+        {
+            return RegistratedTemplates.Select(item => new CommitTemplateItem(item.Key, item.Value())).ToArray();
+        }
+
+        public static bool TryRegistrate(string key, Func<string> item)
+        {
+            if (RegistratedTemplates.ContainsKey(key))
+                return false;
+            RegistratedTemplates.Add(key, item);
+            return true;
+        }
+
+        public static void UnRegistrate(string key)
+        {
+            if (RegistratedTemplates.ContainsKey(key))
+                RegistratedTemplates.Remove(key);
+        }
+
```

Comment:
I don't think `CommitTemplateItem` is the right place to host this functionality. `CommitTemplateItem` is a DTO which is currently overloaded with behavioral concerns (such as `LoadFromSettings()`). This isn't good as this is ultimately adds to the tech debt.
What I propose we do instead is create a separate class (let's call it `CommitTemplateManager`) which will host your functionality

```
    public interface ICommitTemplateManager
    {
        IEnumerable<CommitTemplateItem> RegisteredTemplates { get; }
        void Register(string templateName, Func<string> templateText);
        void Unregister(string templateName);
    }

    public sealed class CommitTemplateManager : ICommitTemplateManager
    {
        private static readonly ConcurrentDictionary<string, Func<string>> RegisteredTemplatesDic = new ConcurrentDictionary<string, Func<string>>();


        public IEnumerable<CommitTemplateItem> RegisteredTemplates => RegisteredTemplatesDic.Select(item => new CommitTemplateItem(item.Key, item.Value()));


        public void Register(string templateName, Func<string> templateText)
        {
            if (RegisteredTemplatesDic.ContainsKey(templateName))
            {
                return;
            }
            RegisteredTemplatesDic.TryAdd(templateName, templateText);
        }

        public void Unregister(string templateName)
        {
            Func<string> notUsed;
            RegisteredTemplatesDic.TryRemove(templateName, out notUsed);
        }
    }
```

This allows a clean separation of concerns, You can now test you changes as well (for example have a look at `GitRemoteControllerTests`).
We then can later move the remaining serialisation logic from `CommitTemplateItem` into this new class.